### PR TITLE
Update to Gnome-40

### PR DESCRIPTION
**Pull request by Gottox**
## Platforms:
* [ ] aarch64-musl
* [ ] aarch64
* [x] armv6l-musl
* [ ] armv6l
* [x] armv7l-musl
* [ ] armv7l
* [ ] i686
* [ ] ppc-musl
* [ ] ppc
* [ ] ppc64-musl
* [ ] ppc64
* [ ] ppc64le-musl
* [ ] ppc64le
* [ ] ppcle-musl
* [ ] ppcle
* [x] x86_64-musl
  * [x] compile
  * [x] check
* [x] x86_64
  * [x] compile
  * [x] check

## TODO
* [ ] update vala
* [ ] update tracker-miners
* [ ] functional test
* [x] python3-gobject tests fail on x86_64
* [x] gspell tests fail on x86_64
* [x] dconf tests fail on x86_64
* [x] glib tests fail on x86_64-musl
* [x] gobject-introspection fails on cross armv7hf

### CODE REVIEW

**Comment by ericonr** [comment]
Might want to compare to #29959 and paper's branch ;)

**Comment by Gottox** [comment]
duplicate of #29959.

**Comment by Oreo639** [comment]
I haven't tested this, but, assuming this works I have no issues with this superseding my pr. (less work for me, and my pr was simply meant to be just enough to be functional)

However I still have a few comments/questions:
You probably want to double check the mutter commit(s) as it seems a little messed up. (also double check the changes in the second mutter commit)

Doesn't gnome-shell still depend on gtk+3 as well or am I mistaken? (I didn't feel like testing since it would have taken too long)
https://github.com/GNOME/gnome-shell/blob/master/meson.build#L84

**Comment by Gottox** [comment]
@Oreo639 I'll rebase on you're PR once I've got the initial webkit2gtk update done.

**Comment by Gottox** [comment]
@q66 waiting till you update webkitgtk and patches.

File: common/shlibs
Comment (Oreo639) [inline common/shlibs:2938]: I'm guessing you want the `.so.0` split apart from the `.so` files but you accidentally left them in above.

File: srcpkgs/mutter/template
Comment (Oreo639) [inline srcpkgs/mutter/template:24]: libmutter-8 doesn't provide `libmutter-cogl-path-8.so`, unless I am missing something.

Comment (Gottox) [reply srcpkgs/mutter/template]: fixed

File: common/shlibs
Comment (Gottox) [reply common/shlibs]: fixed

**Comment by Gottox** [comment]
Blocked by: #29689

**Comment by Gottox** [comment]
gobject-introspection-1.68.0_1 build error on armv7hf:
```
[69/99] Generating gir-glib with a custom command
FAILED: gir/GLib-2.0.gir 
/builddir/gobject-introspection-1.68.0/build/tools/g-ir-scanner --output=gir/GLib-2.0.gir --no-libtool --quiet --reparse-validate --add-include-path /builddir/gobject-introspection-1.68.0/build/gir --add-include-path /builddir/gobject-introspection-1.68.0/gir --use-binary-wrapper=/usr/bin/g-ir-scanner-qemuwrapper --use-ldd-wrapper=/usr/bin/g-ir-scanner-lddwrapper --identifier-prefix=G --symbol-prefix=g --symbol-prefix=glib --c-include=glib.h --namespace=GLib --nsversion=2.0 --library=glib-2.0 --library=gobject-2.0 --external-library --pkg=glib-2.0 --cflags-begin -I/usr/armv7l-linux-gnueabihf/usr/include/glib-2.0 -I/usr/armv7l-linux-gnueabihf/usr/lib32/glib-2.0/include -DGLIB_COMPILATION -D__G_I18N_LIB_H__ -DGETTEXT_PACKAGE=Dummy --cflags-end /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/gobject/glib-types.h /usr/armv7l-linux-gnueabihf/usr/lib32/glib-2.0/include/glibconfig.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib-unix.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/galloca.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/garray.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gasyncqueue.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gatomic.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gbacktrace.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gbase64.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gbitlock.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gbookmarkfile.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gbytes.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gcharset.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gchecksum.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gconvert.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gdataset.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gdate.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gdatetime.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gdir.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/genviron.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gerror.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gfileutils.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/ggettext.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/ghash.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/ghmac.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/ghook.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/ghostutils.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gi18n-lib.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gi18n.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/giochannel.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gkeyfile.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/glist.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gmacros.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gmain.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gmappedfile.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gmarkup.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gmem.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gmessages.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gnode.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/goption.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gpattern.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gpoll.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gprimes.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gprintf.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gqsort.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gquark.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gqueue.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/grand.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/grcbox.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/grefcount.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/grefstring.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gregex.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gscanner.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gsequence.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gshell.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gslice.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gslist.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gspawn.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gstdio.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gstrfuncs.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gstring.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gstringchunk.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gstrvbuilder.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gtestutils.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gthread.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gthreadpool.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gtimer.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gtimezone.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gtrashstack.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gtree.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gtypes.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gunicode.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/guri.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gutils.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/guuid.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gvariant.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gvarianttype.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gversion.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gversionmacros.h /usr/armv7l-linux-gnueabihf/usr/include/glib-2.0/glib/gwin32.h ../gir/glib-2.0.c
Traceback (most recent call last):
  File "/builddir/gobject-introspection-1.68.0/build/tools/g-ir-scanner", line 98, in <module>
    from giscanner.scannermain import scanner_main
  File "/builddir/gobject-introspection-1.68.0/build/giscanner/scannermain.py", line 35, in <module>
    from giscanner.ast import Include, Namespace
  File "/builddir/gobject-introspection-1.68.0/build/giscanner/ast.py", line 29, in <module>
    from .sourcescanner import CTYPE_TYPEDEF, CSYMBOL_TYPE_TYPEDEF
  File "/builddir/gobject-introspection-1.68.0/build/giscanner/sourcescanner.py", line 34, in <module>
    from giscanner._giscanner import SourceScanner as CSourceScanner
ImportError: Error loading shared library /builddir/gobject-introspection-1.68.0/build/giscanner/_giscanner.cpython-39-x86_64-linux-gnu.so: Exec format error
[70/99] Linking target girepository/cmph/cmph-bdz-test
[71/99] Linking target girepository/gthash-test
[72/99] Linking target girepository/libgirepository-1.0.so.1.0.0
ninja: build stopped: subcommand failed.
=> ERROR: gobject-introspection-1.68.0_1: do_build: '${make_cmd} -C ${meson_builddir} ${makejobs} ${make_build_args} ${make_build_target}' exited with 1
=> ERROR:   in do_build() at common/build-style/meson.sh:132
```

**Comment by Gottox** [comment]
# :smile: 
Screenshot from 2021-04-05 21-56-31
# :disappointed: 
Screenshot from 2021-04-05 23-57-59

**Comment by ericonr** [comment]
Have you tried on real hardware? I can give it a test drive.

**Comment by ericonr** [comment]
Btw, please include https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2010 as a backport.

**Comment by Gottox** [comment]
> Have you tried on real hardware? I can give it a test drive.

Not yet. I at least want to have _some_ stability on qemu. Atm it's pretty much unusable on it.

**Comment by Gottox** [comment]
> Btw, please include https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2010 as a backport.
applied
> And I would bump `common/shlibs` for all the GNOME packages as well (there's a high chance they will be using newly available functions from other GNOME pieces).
Good idea

The glib fix has no effect on the stability issues in qemu.

But I see a second issue in dmesg:
```
[  128.022735] gsd-usb-protect[1585]: segfault at 8 ip 000055b7a42fd118 sp 00007fffa0c72190 error 4 in gsd-usb-protection[55b7a42fb000+3000]
```

**Comment by Gottox** [comment]
https://gitlab.gnome.org/GNOME/gnome-settings-daemon/-/issues/585

That looks like my exact issue.

**Comment by Gottox** [comment]
Okay, what I found out so far:

* this issue seems to happen only on VMs so far (at least the issues mention that)
* there's a relation between the session crash and the crash of the gsd service.

The actual session log looks like this:

```
** (gsd-usb-protection:2366): WARNING **: 08:00:25.614: Failed to connect to the screen saver: Error calling StartServiceByName for org.gnome.ScreenSaver: Timeout was reached

(gsd-usb-protection:2366): GLib-GIO-CRITICAL **: 08:00:25.614: g_dbus_proxy_call_sync_internal: assertion 'G_IS_DBUS_PROXY (proxy)' failed
gnome-session-binary[1914]: WARNING: App 'org.gnome.SettingsDaemon.UsbProtection.desktop' respawning too quickly
gnome-session-binary[1914]: WARNING: Application 'org.gnome.SettingsDaemon.UsbProtection.desktop' killed by signal 11

(process:2476): Gtk-WARNING **: 08:00:25.669: Locale not supported by C library.
	Using the fallback 'C' locale.
Window manager warning: Failed to create DRM buffer wrapper: drmModeAddFB does not support format 'AR24' (0x34325241)
Window manager warning: Failed to create DRM buffer wrapper: drmModeAddFB does not support format 'AR24' (0x34325241)

(gnome-shell:1976): mutter-CRITICAL **: 08:00:26.173: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
```

**Comment by Gottox** [comment]
Okay, the issue does not occur on real hardware. There are some input issues on my gpd pocket 2, but I guess that's some upstream bug and the way the screen is rotated on that particular device.

**Comment by Gottox** [comment]
@void-linux/pkg-committers Feel free to review! :1st_place_medal:

**Comment by fosslinux** [comment]
Do we care about QEMU enough to hold of merging until that bug is fixed?

**Comment by Gottox** [comment]
We also need to think about rearranging the dependencies. Installing gnome itself doesn't result in a usable system as the drivers are missing. `xbps-install xorg gnome` installs those drivers, so that's the shortcut I'm using. But I think that should be done in a dedicated PR.

**Comment by Gottox** [comment]
@fosslinux imho having gnome in qemu is not a common usecase. I only do that for testing and verifying.

File: srcpkgs/folks/template
Comment (Piraty) [inline srcpkgs/folks/template:3]: 0.15 is out (not sure if odds are dev releases)

File: srcpkgs/mutter/template
Comment (Piraty) [inline srcpkgs/mutter/template:52]: you could glob as well

**Review by Piraty — CHANGES_REQUESTED** [review]
great work @Gottox, thanks

File: srcpkgs/mutter/template
Comment (Gottox) [reply srcpkgs/mutter/template:52]: I better keep this as it is. I this could result in silent errors.

File: srcpkgs/libvirt-glib/template
Comment (Gottox) [reply srcpkgs/libvirt-glib/template:20]: fixed

File: srcpkgs/libgweather/template
Comment (Gottox) [reply srcpkgs/libgweather/template:11]: set as `depends`.

File: srcpkgs/libgweather/template
Comment (Gottox) [reply srcpkgs/libgweather/template:9]: fixed

File: srcpkgs/gtk4/template
Comment (Gottox) [reply srcpkgs/gtk4/template:35]: fixed

File: srcpkgs/folks/template
Comment (Gottox) [reply srcpkgs/folks/template:3]: afaik folks follows the gnome release style. 0.15 is unstable.

File: srcpkgs/planner/template
Comment (Gottox) [reply srcpkgs/planner/template:3]: It's not really a gnome product, so these updates weren't in scope. But hey, let's do it.

File: srcpkgs/io.elementary.calendar/template
Comment (Gottox) [reply srcpkgs/io.elementary.calendar/template:3]: It's not really a gnome product, so these updates weren't in scope. But hey, let's do it.

Comment (Gottox) [reply srcpkgs/io.elementary.calendar/template:3]: Eh,... There's a bit of patching going on. I'll update that in its own PR.

File: srcpkgs/planner/template
Comment (Gottox) [reply srcpkgs/planner/template]: done

**Comment by Gottox** [comment]
@Piraty changes were applied!

File: srcpkgs/mutter/template
Comment (sgn) [reply srcpkgs/mutter/template:52]: glob has a bug, those files will be moved into `${DESTDIR}/usr/lib/mutter-*/` literally (I've run into this myself).

Comment (Piraty) [reply srcpkgs/mutter/template:52]: agree @Gottox

File: srcpkgs/gnome-books/template
Comment (circhioz) [inline srcpkgs/gnome-books/template:9]: `tracker-devel` here is not needed, `gnome-books` has been ported to `tracker3`

File: srcpkgs/gnome-boxes/template
Comment (circhioz) [inline srcpkgs/gnome-boxes/template:11]: `tracker-devel` here is not needed, `gnome-boxes` has been ported to `tracker3`

File: srcpkgs/gnome-boxes/template
Comment (circhioz) [inline srcpkgs/gnome-boxes/template:13]: Should depend on `tracker3`

**Comment by circhioz** [comment]
`gnome-latex` should be rebuilt for `tepl-6`. There's an upstream commit on master https://gitlab.gnome.org/GNOME/gnome-latex/-/commit/e1b01186f8a4e5d3fee4c9ccfbedd6d098517df9

File: srcpkgs/gnome-books/template
Comment (Gottox) [reply srcpkgs/gnome-books/template:9]: fixed

File: srcpkgs/gnome-boxes/template
Comment (Gottox) [reply srcpkgs/gnome-boxes/template:13]: fixed

File: srcpkgs/gnome-boxes/template
Comment (Gottox) [reply srcpkgs/gnome-boxes/template:11]: fixed

**Comment by Gottox** [comment]
> `gnome-latex` should be rebuilt for `tepl-6`. There's an upstream commit on master https://gitlab.gnome.org/GNOME/gnome-latex/-/commit/e1b01186f8a4e5d3fee4c9ccfbedd6d098517df9

Why not wait for a release?

> `gnome-latex` should be rebuilt for `tepl-6`. There's an upstream commit on master https://gitlab.gnome.org/GNOME/gnome-latex/-/commit/e1b01186f8a4e5d3fee4c9ccfbedd6d098517df9

fixed

**Comment by Gottox** [comment]
Any further blockers? I'll merge around friday evening 18:00 CET.

**Comment by FollieHiyuki** [comment]
@Gottox can you take the change from #29025 into this as well and see if it works?

**Comment by Gottox** [comment]
> @Gottox can you take the change from #29025 into this as well and see if it works?

Let me remove this commit. It makes more sense to update it independant of gnome.

**Comment by ericonr** [comment]
```
=> mutter-40.0_1: fetching distfile 'mutter-40.0.tar.xz'...
https://download.gnome.org/sources/mutter/40.0/mutter-40.0.tar.xz: Not Found
```

Doesn't seem to be due to load limiting, I can't access it locally either.

There are some packages with up to 3 commits, might be nice to squash them together...

**Comment by Gottox** [comment]
> ```
> => mutter-40.0_1: fetching distfile 'mutter-40.0.tar.xz'...
> https://download.gnome.org/sources/mutter/40.0/mutter-40.0.tar.xz: Not Found
> ```
> 
> Doesn't seem to be due to load limiting, I can't access it locally either.
It was a real issue. Fixed.

> There are some packages with up to 3 commits, might be nice to squash them together...
commits squashed.

**Comment by thegarlynch** [comment]
Wait, gnome 40 is not different package ? i thought it is different in windows 8 and windows 10 kind of way

**Comment by Oreo639** [comment]
GNOME is a desktop environment and Windows is an operating system, so this isn't a great comparison.
Void Linux is a rolling release distro meaning that there is only one version of Void Linux and that is constantly being updated with the latest versions of packages (with patches and bugfixes if necessary).

Gnome 40 is a new version of the GNOME desktop environment and is not a new DE. (it is meant to replace Gnome 3.38)

There is a versioning scheme that exists with other "stable" distros (which just means that the packages stay the same for that release of the distro and only get patched with bugfixes and security fixes).

**Comment by Gottox** [comment]
There are several changes in the frontend, but from a packaging perspective it's very much like a Gnome 3.40.

**Comment by thegarlynch** [comment]
Holyshit, it's great. Did they improved the performance or something ? it runs smoother compared to previous version of gnome i've run

**Pull request merged**
